Source/Diagnostics/ReducedDiags/BeamRelevant.cpp

```diff
@@ -242,18 +243,17 @@ void BeamRelevant::ComputeDiags (int step)
                         p_ux_mean, p_uy_mean, p_uz_mean,
                         p_gm_mean};
             },
-            reduce_ops);
-
-        std::vector<ParticleReal> values_per_rank_1st = {
-            amrex::get<0>(r), // w
-            amrex::get<1>(r), // x_mean
-            amrex::get<2>(r), // y_mean
-            amrex::get<3>(r), // z_mean
-            amrex::get<4>(r), // ux_mean
-            amrex::get<5>(r), // uy_mean
-            amrex::get<6>(r), // uz_mean
-            amrex::get<7>(r), // gm_mean
-        };
+            reduce_ops_1);
+
+        std::vector<ParticleReal> values_per_rank_1st(num_red_ops_1);
+
+        /* contains in this order:
+         * w, x_mean, y_mean, z_mean
+         * ux_mean, uy_mean, uz_mean, gm_mean
+         */
+        amrex::constexpr_for<0, num_red_ops_1> ([&](auto i) {
+            values_per_rank_1st[i] = amrex::get<i>(r1);
```

Comment:
For some reason, @AlexanderSinn uses this syntax in HiPACE++:
```suggestion
values_per_rank_1st[i.value] = amrex::get<i.value>(r1);
```